app/src/main/java/org/zephyrsoft/trackworktime/timer/TimeCalculatorV2.java

```diff
@@ -247,7 +247,7 @@ private long calculateWorkTime(List<Event> events) {
 				timeOut = currentDate.atTime(LocalTime.MAX).atZone(zoneId).toOffsetDateTime();
 
 				workedTime += TimerManager.timeDiff(clockedInSince, timeOut);
-			} else if (effectiveClockOutEvent != null) {
```

Comment:
I'd like to keep this here. If `effectiveClockOutEvent` isn't `null`, it does not harm. And if it is `null`, it prevents a NPE I saw multiple times in crash reports from users.

Reply:
It harms to that extent that you don't get any feedback that the logic is faulty and the calculation is wrong. If an exception is thrown, parts of the week table stay blank, so this is a clear indication for the user, that something is wrong.  But that is your decision. I can change the pull request.

Reply:
I'd have no problem with a part of the table being empty, but in the cases for which I received reports the whole app had crashed. Maybe a `try` around the whole time calculation (with at least a WARN log output in the `catch` block) would be appropriate?

Reply:
The calculation for the week table is inside ``try`` - ``catch``:
https://github.com/mathisdt/trackworktime/blob/cb59ed11df5a2c8152db7d22dab9457ca9e1a5dd/app/src/main/java/org/zephyrsoft/trackworktime/weektimes/WeekStateCalculator.java#L68

But, I didn't think of the widget. I agree that should be the case with the widget as well. I will update the pull request, but probably not today...

Reply:
I have removed the revert and added a comment on why the null check is necessary.